pkg/test/runner/runner.go

```diff
@@ -206,11 +202,8 @@ func (r *Runner) runFnRender() error {
 	for i := 0; i < r.testCase.Config.RunCount(); i++ {
 		command.SetArgs(kptArgs)
 		fnErr = command.Execute()
-		if fnErr != nil {
-			if r.testCase.Config.ExitCode != 0 {
-				return nil
-			}
-			break
+		if fnErr != nil && r.testCase.Config.ExitCode != 0 {
+			return nil
```

Comment:
The previous logic was stopping the for loop for the case where command returned the error and expected ExitCode was 0 (success), but now it will continue with the loop (which we don't want..right ?)

Reply:
I will mark this PR as draft. We will need to merge #1747 at first.